sdks/java/fn-execution/src/main/java/org/apache/beam/sdk/fn/data/BeamFnDataTimeBasedBufferingOutboundObserver.java

```diff
@@ -48,25 +46,27 @@
       Coder<T> coder,
       StreamObserver<BeamFnApi.Elements> outboundObserver) {
     super(sizeLimit, outputLocation, coder, outboundObserver);
-    this.lock = new Object();
+    this.flushLock = new Object();
     this.flushFuture =
         Executors.newSingleThreadScheduledExecutor(
                 new ThreadFactoryBuilder()
                     .setDaemon(true)
                     .setNameFormat("DataBufferOutboundFlusher-thread")
                     .build())
             .scheduleAtFixedRate(this::periodicFlush, timeLimit, timeLimit, TimeUnit.MILLISECONDS);
```

Comment:
You should be able to use the callable version of schedule:
https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ScheduledExecutorService.html#schedule-java.util.concurrent.Callable-long-java.util.concurrent.TimeUnit-
to invoke the flush method. You might need to provide a cast to Callable<Void> to tell Java to use that version of schedule.

Reply:
I found that the callable version of schedule is not a periodic action. So, we have to create a new method which warps the `flush`, What do you think?

Reply:
Your right, I missed the fact that there was no periodic schedule that took a callable.